### PR TITLE
Improve ZarrSortingExtractor

### PR DESCRIPTION
**Pull request by samuelgarcia**
Add option for ZarrSortingExtractor to read in sub group.

### CODE REVIEW

File: src/spikeinterface/core/tests/test_zarrextractors.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ZarrSortingExtractor,
+    generate_sorting,
+    load_extractor,
+)
+from spikeinterface.core.zarrextractors import add_sorting_to_zarr_group
+
+if hasattr(pytest, "global_test_folder"):
+    cache_folder = pytest.global_test_folder / "core"
+else:
+    cache_folder = Path("cache_folder") / "core"
```
Comment (alejoe91) [inline src/spikeinterface/core/tests/test_zarrextractors.py:19]: Can you use `tmp_path` fixture instead?

File: src/spikeinterface/core/tests/test_zarrextractors.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    folder = cache_folder / "zarr_sorting_sub_group"
+    if folder.is_dir():
+        shutil.rmtree(folder)
+    zarr_root = zarr.open(folder, mode="w")
+    zarr_sorting_group = zarr_root.create_group("sorting")
+    add_sorting_to_zarr_group(sorting, zarr_sorting_group)
+    sorting = ZarrSortingExtractor(folder, zarr_group="sorting")
+    # and reaload
+    sorting = load_extractor(sorting.to_dict())
+
```
Comment (alejoe91) [inline src/spikeinterface/core/tests/test_zarrextractors.py:43]: check if they are the same?

Comment (samuelgarcia) [reply src/spikeinterface/core/tests/test_zarrextractors.py:43]: uselless I think, this already catch the kwargs stuff no ?

File: src/spikeinterface/core/tests/test_zarrextractors.py (hunk 1 above)
Comment (samuelgarcia) [reply src/spikeinterface/core/tests/test_zarrextractors.py:19]: no later and everywhere

File: src/spikeinterface/core/zarrextractors.py [hunk 3]
```diff
@@ -215,7 +221,7 @@ def __init__(self, folder_path: Path | str, storage_options: dict | None = None)
         if annotations is not None:
             self.annotate(**annotations)
 
-        self._kwargs = {"root_path": folder_path_kwarg, "storage_options": storage_options}
+        self._kwargs = {"folder_path": folder_path_kwarg, "storage_options": storage_options, "zarr_group": zarr_group}
```
[the remark this answers is not in the document]
Comment (samuelgarcia) [reply src/spikeinterface/core/zarrextractors.py:224]: but easy when we have a test file for this !!

File: src/spikeinterface/core/zarrextractors.py [hunk 4]
```diff
@@ -154,11 +154,13 @@ class ZarrSortingExtractor(BaseSorting):
...
     ----------
-    root_path: str or Path
+    folder_path: str or Path
         Path to the zarr root file
     storage_options: dict or None
         Storage options for zarr `store`. E.g., if "s3://" or "gcs://" they can provide authentication methods, etc.
-
+    zarr_group: None | str
+        Optionaly a zarr group when the sorting is not stored at the root but in sub group.
+        Usefull for more complex storage that store SOrting in sub group like the futur SortingResult.
```
Comment (alejoe91) [inline src/spikeinterface/core/zarrextractors.py:163]:
```suggestion
    zarr_group: str or None, default: None
        Optional zarr group when the sorting is not stored at the root, but in sub group.
```

File: src/spikeinterface/core/tests/test_zarrextractors.py (hunk 1 above)
Comment (alejoe91) [reply src/spikeinterface/core/tests/test_zarrextractors.py:19]: ok

**Comment by samuelgarcia** [comment]
Merci camarade.

**Pull request merged**